source/containers.tex

```diff
@@ -16086,6 +16093,1215 @@
+\tcode{uses_allocator_v<container_type, Allocator>} is \tcode{true}.
+
+\pnum
+\effects
+Equivalent to \tcode{flat_set(s, cont)},
+except that \exposid{c} is constructed with
+uses-allocator construction\iref{allocator.uses.construction}.
+
+\pnum
+\complexity
+Linear.
+\end{itemdescr}
+
+\indexlibraryctor{flat_set}%
+\begin{itemdecl}
+template<class Allocator>
+  flat_set(const key_compare& comp, const Allocator& a);
+template<class Allocator>
+  explicit flat_set(const Allocator& a);
+template<class InputIterator, class Allocator>
+  flat_set(InputIterator first, InputIterator last,
+           const key_compare& comp, const Allocator& a);
+template<class InputIterator, class Allocator>
+  flat_set(InputIterator first, InputIterator last, const Allocator& a);
+template<@\exposconcept{container-compatible-range}@<value_type> R, class Allocator>
+  flat_set(from_range_t, R&& rg, const Allocator& a);
+template<@\exposconcept{container-compatible-range}@<value_type> R, class Allocator>
+  flat_set(from_range_t, R&& rg, const key_compare& comp, const Allocator& a);
+template<class InputIterator, class Allocator>
+  flat_set(sorted_unique_t, InputIterator first, InputIterator last,
+           const key_compare& comp, const Allocator& a);
+template<class InputIterator, class Allocator>
+  flat_set(sorted_unique_t, InputIterator first, InputIterator last, const Allocator& a);
+template<class Allocator>
+  flat_set(initializer_list<key_type> il, const key_compare& comp, const Allocator& a);
+template<class Allocator>
+  flat_set(initializer_list<key_type> il, const Allocator& a);
+template<class Allocator>
+  flat_set(sorted_unique_t, initializer_list<key_type> il,
+           const key_compare& comp, const Allocator& a);
+template<class Allocator>
+  flat_set(sorted_unique_t, initializer_list<key_type> il, const Allocator& a);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\constraints
+\tcode{uses_allocator_v<container_type, Allocator>} is \tcode{true}.
+
+\pnum
+\effects
+Equivalent to the corresponding non-allocator constructors
+except that \exposid{c} is constructed with
+uses-allocator construction\iref{allocator.uses.construction}.
+\end{itemdescr}
+
+\rSec3[flat.set.modifiers]{Modifiers}
+
+\indexlibrarymember{insert}{flat_set}%
+\begin{itemdecl}
+template<class K> pair<iterator, bool> insert(K&& x);
+template<class K> iterator insert(const_iterator hint, K&& x);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\constraints
+The \grammarterm{qualified-id} \tcode{Compare::is_transparent}
+is valid and denotes a type.
+\tcode{is_constructi\-ble_v<value_type, K>} is \tcode{true}.
+
+\pnum
+\expects
+The conversion from \tcode{x} into \tcode{value_type} constructs
+an object \tcode{u}, for which \tcode{find(x) == find(u)} is true.
+
+\pnum
+\effects
+If the set already contains an element equivalent to \tcode{x},
+\tcode{*this} and \tcode{x} are unchanged.
+Otherwise,
+inserts a new element as if by: \tcode{return emplace(std::forward<K>(x));}.
```

Comment:
```suggestion
inserts a new element as if by: \tcode{emplace(std::forward<K>(x))}.
```

Reply:
In the original paper, I think,.

Reply:
The `return` seems useful here?

Reply:
I doubt it. But perhaps the addition of result here was intentional. But I don't think it makes sense. First of all, this has a _Returns_ element. And secondly, precedes the piece of code is "inserts a new element as if by", which suggests it's not specifying (part of) the result value.

Reply:
Finally, the _Returns_ element fully describes the result.